### DISCUSSION TITLE
Automate TeamTalk server configuration file creation

### DISCUSSION THREAD

**Discussion by tech10** [BearWare/TeamTalk5#1117] [Ideas]
Hello,
I've noticed an increasing number of TeamTalk server hosting providers, but I'm not sure they automatically create configuration files. This could be done with a script or other programming language, of course, but I'm curious to know what everyone else thinks of this idea. This could create a TeamTalk configuration file silently and automatically in the working directory, for example.

Here is an example command line that could be created for such a purpose.

```console
$ tt5srv --wizard-silent --config-ip :: --config-tcp 10333 --config-udp 10333 --config-user-name admin --config-user-password test --config-user-admin true
```

I would personally find this useful if I was to set up a free TeamTalk server hosting service, which could automatically provision servers for users, etc. Thoughts, ideas?

**Comment by bear101** [comment]
I would suggest you use Python to generate the tt5srv.xml file. You can use [TeamTalk5.py](https://github.com/BearWare/TeamTalk5/blob/master/Library/TeamTalkPy/TeamTalk5.py) to generate the correct [user-type](https://github.com/BearWare/TeamTalk5/blob/master/Library/TeamTalkPy/TeamTalk5.py#L537) and [user-right](https://github.com/BearWare/TeamTalk5/blob/master/Library/TeamTalkPy/TeamTalk5.py#L447).

**Comment by tech10** [comment]
This is certainly an option, but if I understand the code correctly, this would require a license for the TeamTalk SDK. Also, it would require the use of Python, whereas my idea would only require whatever programming language you're using, to execute the TeamTalk server binary, which could generate the configuration file for you silently. Also, it would automatically support each platform that the TeamTalk server supports, something the Python library doesn't yet do. Or, would require more work to support the alternative platforms.

**Comment by bear101** [comment]
You don't need a TeamTalk SDK license if you only use TeamTalk5.py for generating XML files. It's only if you instantiate the TeamTalk shared library that you need a SDK license.

**Comment by tech10** [comment]
I would still consider it to be a useful feature to have in the server binary, even if it simply generated a configuration file with default values that could then be altered, with a minimal set of parameters you could include for silent configuration generation, an IP address, port, and the username and password of one administrator account.

I've considered creating a tool for this in Go, which can be statically compiled and executed on any platform that Go supports. If there is any interest in this from anyone, feel free to reply here and I'll work on it, and, should I work on this in the mean time, I'll be sure to post a link to it.
